craft_parts/executor/part_handler.py

```diff
@@ -248,6 +257,19 @@ def _run_step(
 
         return step_handler.run_builtin()
 
+    def _compute_layer_hash(self, *, for_all_parts: bool = False) -> LayerHash:
```

Comment:
you seem to always set this to True, is there a reason for this parameter?

Reply:
The overlay step handler uses the validation hash for layers up to the step being processed (but this handler is not implemented yet).

Reply:
Added a docstring for `_compute_layer_hash()`.